File: palindrome_string.py
def isPalindrome(string):
    left, right = 0, len(string)-1
    while left < right:
        if string[left] != string[right]:
            return False
        left += 1
        right -= 1

    return True

if __name__ == "__main__":
    s = "abccba"
    print("Is palindrome?", isPalindrome(s))

# output: ('Is palindrome?', True)

File: palindrome_string.go
package main

import "fmt"

func isPalindrome(s string) bool {
	left, right := 0, len(s)-1
	for left < right {
		if s[left] != s[right] {
			return false
		}
		left += 1
		right -= 1
	}
	return true
}

func main() {
	s := "abccba"
	fmt.Println("Is palindrome?", isPalindrome(s))
}

// output: Is palindrome? true
